certifikit-cli/src/main/kotlin/app/cash/certifikit/cli/certificateOutput.kt

```diff
@@ -98,6 +100,12 @@ private fun Certificate.subjectAlternativeNameValue(): List<String>? {
         it.value as List<Pair<Any, Any>>
       }
       ?.map {
-        it.second.toString()
+        if (it.second is String) {
+          it.second as String
+        } else if (it.first is BasicDerAdapter<*> && (it.first as BasicDerAdapter<*>).tag == 7L) {
```

Comment:
This awkwardly needs to match against CertificateAdapters.kt

```
  internal val generalNameDnsName = Adapters.IA5_STRING.withTag(tag = 2L)
  internal val generalNameUniformResourceIdentifier = Adapters.IA5_STRING.withTag(tag = 6L)
  internal val generalNameIpAddress = Adapters.OCTET_STRING.withTag(tag = 7L)
  internal val generalName: DerAdapter<Pair<DerAdapter<*>, Any?>> = Adapters.choice(
      generalNameDnsName,
      generalNameUniformResourceIdentifier,
      generalNameIpAddress,
      Adapters.ANY_VALUE
  )
```

Reply:
Any advice on restructuring both side of this happily accepted.

Reply:
Extracting first and second and giving them names would be better probably

```
?.map { (first_with_meaningful_name, second_with_meaningful_name) ->
}
```